dogm/demo/utils/image_creation.cpp

```diff
@@ -178,3 +178,53 @@ cv::Mat compute_particles_image(const dogm::DOGM& grid_map)
 
     return particles_img;
 }
+
+static void add_text(const std::string& text, const float relative_horizontal_position, cv::Mat& img)
+{
+    int fontFace = cv::FONT_HERSHEY_DUPLEX;
+    double fontScale = 1;
+    int thickness = 1;
+    int baseline{0};
+    cv::Size textSize = cv::getTextSize(text, fontFace, fontScale, thickness, &baseline);
+    cv::Point textOrg((img.cols - textSize.width) * relative_horizontal_position, (img.rows + textSize.height) * 0.5f);
+    cv::putText(img, text, textOrg, fontFace, fontScale, cv::Scalar::all(255), thickness, 8);
+}
+
+void computeAndSaveResultImages(const dogm::DOGM& grid_map,
+                                const std::vector<Point<dogm::GridCell>>& cells_with_velocity, const int step,
+                                const bool concatenate_images, const bool show_during_execution)
+{
+    cv::Mat raw_meas_grid_img = compute_raw_measurement_grid_image(grid_map);
+    cv::Mat particle_img = compute_particles_image(grid_map);
+    cv::Mat dogm_img = compute_dogm_image(grid_map, cells_with_velocity);
+
+    cv::Mat image_to_show{};
+    if (concatenate_images)
+    {
+        cv::hconcat(dogm_img, particle_img, image_to_show);
+        cv::hconcat(image_to_show, raw_meas_grid_img, image_to_show);
+
+        cv::Mat subtitles(image_to_show.rows * 0.2f, image_to_show.cols, image_to_show.type(), cv::Scalar::all(30));
+        add_text("Grid", 0.13f, subtitles);
```

Comment:
If you apply the text to the images before concatenating them, then the text position would be the same for all images. Wouldn't that be simpler?

Reply:
True, good idea! Implemented.